common/lobbies/slot.ts

```diff
@@ -17,7 +28,7 @@ export class Slot extends Record({
 
 export function createOpen(race: RaceChar = 'r', hasForcedRace = false, playerId = 0): Slot {
   return new Slot({
-    type: 'open',
+    type: SlotType.Open as typeof SlotType.Open,
```

Comment:
These casts are nonsensical. You're treating these like they're a union type, but there is no union type. There are no subclasses of `Slot`. Every one of these functions returns just a `Slot`, there is nothing to discriminate.